ui/components/Recurrence/RecurrenceSelect.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react';
+import Select, { ActionMeta, FormatOptionLabelMeta, OnChangeValue, Theme } from 'react-select';
+
+import { RecurrenceSelectionOption, SelectOption } from './RecurrenceSelectOption';
+
+import getRecurrencesForDate from 'components/Recurrence/getRecurrencesForDate';
+import Recurrence from 'components/Recurrence/Recurrence';
+
+interface Props<T extends HTMLElement>{
+  // TODO Add a way to pass a current value to the RecurrenceList component.
+  className?: string;
+  menuRef?: T;
+  date: moment.Moment;
+  onChange: { (value: Recurrence): void };
+  disabled?: boolean;
+}
+
+export default function RecurrenceSelect<T extends HTMLElement>(props: Props<T>): JSX.Element {
+  const [selectedIndex, setSelectedIndex] = useState<number | null>(null);
+  const rules = getRecurrencesForDate(props.date);
+
+  function handleRecurrenceChange(newValue: OnChangeValue<SelectOption, false>, _: ActionMeta<SelectOption>) {
+    const { onChange } = props;
+    setSelectedIndex(newValue.value);
+    onChange(rules[newValue.value]);
+  }
+
+  function formatOptionsLabel(option: SelectOption, meta: FormatOptionLabelMeta<SelectOption>): React.ReactNode {
+    if (meta.context === 'value') {
+      return option.label;
+    }
+    return option.label;
+  }
+
+  const options = rules.map((item, index) => ({
+    label: item.name,
+    value: index,
+  }));
+
+  const ref = props?.menuRef || document.body;
+  const value = selectedIndex !== null && selectedIndex >= 0 ?
+    options[selectedIndex] :
+    { label: 'Select a frequency...', value: -1 };
```

Comment:
If the props.date provided is the 15th or last day of the month, then there will be additional "options".

If the user selects the last option, but then changes the date they selected it will cause an index out of range error here.